lib/plugindata/access_request.go

```diff
@@ -0,0 +1,61 @@
+package plugindata
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ResolutionTag represents enum type of access request resolution constant
+type ResolutionTag string
+
+const (
+	Unresolved       = ResolutionTag("")
+	ResolvedApproved = ResolutionTag("APPROVED")
+	ResolvedDenied   = ResolutionTag("DENIED")
+	ResolvedExpired  = ResolutionTag("EXPIRED")
+)
+
+// AccessRequestData represents generic plugin data required for access request processing
+type AccessRequestData struct {
+	User             string
+	Roles            []string
+	RequestReason    string
+	ReviewsCount     int
+	ResolutionTag    ResolutionTag
+	ResolutionReason string
```

Comment:
:nail_care: 
The other plugins have those two fields in a single struct - `Resolution`
https://github.com/gravitational/teleport-plugins/blob/7ea82ce5d94119aeb5f23a40a8fe88edbed3d326/access/slack/plugindata.go#L14-L17
It would be good if we did that here as well, so that latter on, we can remove each individual implementation to this `/lib/plugindata`

Reply:
My guess was that the separate Resolution struct her is redundant since it is almost never used separately. `/lib/plugindata` raises lots of questions, so, let me give one more pass to it and then we decide.